content/en/_index.md

```diff
@@ -271,6 +387,13 @@ requests or if they are thread-safe).
      // make a request with the client
    }
 {{< /tab >}}
+  {{< tab header="Node.js" >}}
+    // Imports the Google Cloud Data Loss Prevention library
+    const DLP = require('@google-cloud/dlp');
+
+    // Instantiates a reusable client object
```

Comment:
Does anything need to be done to "clean up" the client? Is it threadsafe?

Reply:
No - and Node is (from the programmer's standpoint) single-threaded.

Reply:
Can we update the comment to reflect that resources used by the client in the background are automatically cleaned up when it is garbage collected?

Is Node.JS _always_ single threaded? Python is _normally_ single threaded, but the ability to spawn different thread/processes exists. [It looks like Node offers something similar](https://nodejs.org/api/worker_threads.html)?

Reply:
This looks unresolved still?

Reply:
I thought about this a bit more - usually, these objects are stored globally and destroyed when the program itself terminates. I'm not sure if they support programmatic garbage collection (as I've never needed to do that).

Is that an OK approach to recommend, or should I check to see if they're properly GC'ed (i.e. don't leak memory when going out of scope)?

Reply:
The two things that I've seen come up in Java and Python are:
1. Is this thing threadsafe?
2. How do I clean up background resources?

I know that _normally_ we tend to recommend global usage, but there seem to almost certainly users that have questions about those 2 things, so some advice on how they should handle them would be good. 

Maybe Ben has some insights into both?

Reply:
I think even if the answer is "These hang onto resources until the process ends", we try to say that as well.

Reply:
I think bcoe or alexander-fenster could help us clarify this.

Reply:
> Is Node.JS always single threaded?

Node.js doesn't have thread primitives like other programming languages, such that you would have to worry about contention on a global instance of the client.

There are [workers](https://nodejs.org/api/worker_threads.html#worker-threads) but they only communicate through a message passing API.

We recommend for environments like Cloud Functions that users create and cache a client.

> Does anything need to be done to "clean up" the client?

Nothing should need to be done to cleanup the client. Coming back to the Cloud Run or Cloud Function example, it's recommended you keep a cached version of the client in the global scope, to reduce startup overhead on subsequent requests. This client will not be GCd until the function shuts down.

If you did create a client on each request, it will be GCd as soon as its enclosing function is done executing.

Reply:
> I think even if the answer is "These hang onto resources until the process ends", we try to say that as well.

It will hang on to resources in memory until it has no reference; if you create the client in the global scope, this will be process termination, if you create it in a closure, this will be when that closure finishes execution.

Reply:
Thanks Ben, sounds like the answer on "do we close/cleanup the client explicitly" is a "no".

For reusing clients across inbound requests, we skirt the edges of best practices in the guidelines, we could perhaps have a comment recommending the client object have global scope. I don't think we want to make it an explicit guideline for all samples.

Reply:
Used the following as a comment:
```
// Instantiates a reusable client object in the global scope.
// (Node.js will automatically clean it up when the script terminates.)
```